src/IconToggle/index.js

```diff
@@ -1,89 +1,120 @@
-      // TODO: Handle when it doesn't properly update. To do this, you would make a setInterval
-      // function to wait 2s (or something similar) and then check if the props did successfully change.
-      const newValue = !localValue
-      localChanges.push(newValue)
-      setLocalChanges(localChanges)
-      await onChange(newValue)
-      if (activeActions && newValue) {
-        await activeActions()
-      }
-      if (inactiveActions && !newValue) {
-        await inactiveActions()
-      }
-      if (clickActions) {
-        await clickActions()
-      }
+  componentDidUpdate(prevProps, prevState) {
+    const { input } = this.props
+
+    if (prevProps.input.value !== input.value) {
+      this.handleLocalValue()
     }
   }
 
-  useEffect(() => {
+  getLocalValue = () => {
+    const { localChanges } = this.state
+    const { input } = this.props
+
+    return localChanges.length !== 0
+      ? localChanges[localChanges.length - 1]
+      : input.value
+  }
+
+  handleLocalValue = () => {
+    let { localChanges } = this.state
+    const { input } = this.props
+
     if (localChanges.length !== 0) {
-      // There are local changes queued up.
-      if (!!value === localChanges[0]) {
+      if (!!input.value !== localChanges[0]) {
+        // remove first item from array
         localChanges.shift()
-        setLocalChanges(localChanges)
+
+        this.setState((state) => ({
+          ...state,
+          localChanges,
+        }))
       }
     }
-  }, [value])
-
-  const styles = {
-    wrapper: {
-      height: toggleSize,
-      width: toggleSize,
-    },
-    buttonWrapper: {
-      margin: -12,
-      width: 2 * toggleSize,
-      height: 2 * toggleSize,
-    },
   }
 
-  const iconName = localValue ? activeIcon : inactiveIcon
-  const iconColor = localValue ? activeColor : inactiveColor
-
-  return (
-    <View style={(styles.wrapper, styles.buttonWrapper)}>
-      <IconToggle
-        name={iconName}
-        color={iconColor}
-        underlayColor={activeColor}
-        maxOpacity={0.3}
-        size={toggleSize}
-        onPress={handlePress}
-        key={`iconToggle.${toggleSize}`}
-      />
-    </View>
-  )
+  handlePress = async () => {
+    const { activeActions, clickActions, inactiveActions, input } = this.props
+
+    if (input.onChange) {
+      const value = !this.getLocalValue()
+
+      // update state with new value
+      this.setState((state) => ({
+        ...state,
+        localChanges: [...state.localChanges, value],
+      }))
+
+      await input.onChange(value)
+
+      if (typeof activeActions === 'function' && value) {
+        await activeActions().catch((err) => {
+          console.error('active actions failed', err)
+        })
+      }
+
+      if (typeof inactiveActions === 'function' && !value) {
+        await inactiveActions().catch((err) => {
+          console.error('inactive actions failed', err)
+        })
+      }
+
+      if (typeof clickActions === 'function') {
```

Comment:
Have you tested to make sure that `typeof clickActions` doesn't return `object`?